build/calculate-cqs.sh

```diff
@@ -13,19 +13,18 @@ mvn dependency:tree -Dstyle.color=never > build/deps-raw
 grep ':compile' build/deps-raw | grep -v 'omitted' | grep -v 'org.locationtech.geomesa' | sed -e 's/\[INFO\] //' -e 's/[\| +-]*//' -e 's/(.*)//' -e 's/ //g' -e 's/\(.*\):\(.*\):jar:\(.*\):\(\w*\)/\1:\2\t\3\t\4/' | sort | uniq > build/cqs.tsv
 echo "" >> build/cqs.tsv
 for cq in $(grep ':provided' build/deps-raw | grep '^\[INFO\] +-' | grep -v 'org.locationtech.geomesa' | sed -e 's/\[INFO\] +- //' -e 's/(.*)//' | sort | uniq); do
-  dep="$(echo $cq | sed 's/\(.*\):\(.*\):jar:\(.*\):\(\w*\)/\1:\2\t\3\t\4/')"
-  reg=$(echo "${dep%	*}" | sed 's/\s\s*/\\s*/g')
-  if [ -z "$(grep $reg build/cqs.tsv)" ]; then
+  dep="$(echo "$cq" | sed -E 's/(.*):(.*):jar:(.*):(\w*)/\1:\2\t\3\t\4/')"
+  reg=$(echo "${dep%	*}" | sed -E 's/\s\s*/\\s*/g')
+  if ! grep -q "$reg" build/cqs.tsv; then
     echo "$dep" >> build/cqs.tsv
   fi
 done
 echo "" >> build/cqs.tsv
 for cq in $(grep ':test' build/deps-raw | grep '^\[INFO\] +-' | grep -v 'org.locationtech.geomesa' | sed -e 's/\[INFO\] +- //' -e 's/(.*)//' | sort | uniq); do
-  dep="$(echo $cq | sed 's/\(.*\):\(.*\):jar:\(.*\):\(\w*\)/\1:\2\t\3\t\4/')"
-  reg=$(echo "${dep%	*}" | sed 's/\s\s*/\\s*/g')
-  if [ -z "$(grep $reg build/cqs.tsv)" ]; then
+  dep="$(echo "$cq" | sed -E 's/(.*):(.*):jar:(.*):(\w*)/\1:\2\t\3\t\4/')"
+  reg=$(echo "${dep%	*}" | sed -E 's/\s\s*/\\s*/g')
```

Comment:
I don't think so... according to [this](https://www.gnu.org/software/sed/manual/html_node/Extended-regexps.html) it only affects `?`, `+`, `()` and `{}`. I think I just was trying to be consistent, and `-E` makes sed behave similar to most other regex engines where you have to escape those characters to treat them as literals (vs the opposite), so it seems like adding it everywhere might make things a little more consistent...

Reply:
actually i remember now... the shellcheck throws out a warning to `consider using ${arg/match/replacement} instead of piping to sed` - but if you use `sed -E` then it doesn't complain :P

in this case the variable replacement doesn't work, so we have to use sed. could alternatively mark the issue as "resolved" in github so it doesn't complain i guess.